internal/services/cdn/cdn_frontdoor_security_policy_resource.go

```diff
@@ -0,0 +1,291 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceCdnFrontDoorSecurityPolicy() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceCdnFrontdoorSecurityPolicyCreate,
+		Read:   resourceCdnFrontdoorSecurityPolicyRead,
+		Delete: resourceCdnFrontdoorSecurityPolicyDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.FrontDoorSecurityPolicyID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+				// TODO: validation
+				// WS: There are literally no rules for what this string can be. Fixed
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"cdn_frontdoor_profile_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FrontDoorProfileID,
+			},
+
+			"security_policies": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+
+						"firewall": {
+							Type:     pluginsdk.TypeList,
+							Required: true,
+							ForceNew: true,
+							MaxItems: 1,
+
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+
+									"cdn_frontdoor_firewall_policy_id": {
+										Type:     pluginsdk.TypeString,
+										Required: true,
+										ForceNew: true,
+										// TODO: validation for the ID type
+										// WS: Fixed
+										ValidateFunc: validate.FrontDoorFirewallPolicyID,
+									},
+
+									"association": {
+										Type:     pluginsdk.TypeList,
+										Required: true,
+										ForceNew: true,
+										MaxItems: 1,
+
+										Elem: &pluginsdk.Resource{
+											Schema: map[string]*pluginsdk.Schema{
+
+												// NOTE: The max number of domains vary depending on sku: 100 Standard, 500 Premium
+												"domain": {
+													Type:     pluginsdk.TypeList,
+													Required: true,
+													ForceNew: true,
+													MaxItems: 500,
+
+													Elem: &pluginsdk.Resource{
+														Schema: map[string]*pluginsdk.Schema{
+
+															// WS: I changed this to a generic name from 'cdn_frontdoor_custom_domain_id' because
+															// this value can be either a custom domain id or a endpoint id...
```

Comment:
and here, I think the validation function is explanatory for valid ID types?
```suggestion
```

Reply:
Fixed.